lib/params.go

```diff
@@ -11,11 +11,16 @@ import (
 // Limit param is provided to a paginated method
 const DefaultPageSize = 100
 
+// BaseParams defines the basic HTTP params for all requests
+type BaseParams interface {
+	Proxied() bool
```

Comment:
I'll admit to being confused about what we intend to do with this. Do we need to know the request is proxied?